libs/designer/src/lib/core/utils/outputs.ts

```diff
@@ -341,7 +341,18 @@ export const getUpdatedManifestForSchemaDependency = (manifest: OperationManifes
       }
 
       const currentSchemaValue = getObjectPropertyValue(updatedManifest.properties.outputs, outputLocation);
-      safeSetObjectPropertyValue(updatedManifest.properties.outputs, outputLocation, { ...currentSchemaValue, ...schemaToReplace });
+
+      // if schema contains static object returned from RP, merge the current schema value and new schema value
+      if (schemaToReplace && 'rows' in schemaToReplace) {
```

Comment:
I personally don't think using a hardcoded property name is reliable to have custom logic for. But don't have a suggestion cause I don't know much about the scenario. Perhaps LAUX team can comment